### ISSUE TITLE
Release first beta of version 12.0

### ISSUE THREAD

**Issue by pedrovereza** [Moya/Moya#1709]
It seems that after #1668 we're ready to release the first beta version for version 12 🎉 

There is, however, issue #1126 still tagged in version 12 milestone that is not resolved. This one has been around for over an year now and I feel it'll be pushed to version 13 (even though #1502 seems to cover part of the job already).

I think we can go ahead with the beta version of 12.0 and #1126 can make it to the final release if it gets resolved in the next weeks.

@Moya/contributors @Moya/core-team Any thoughts on that? 🤔

**Comment by SD10** [comment]
Thanks for raising this issue 💯 I didn't get time to work on the 12.0 release last weekend unfortunately. I plan on trying again this weekend but if anyone wants to work on it before then -- feel free.

Since #1126 is my pr, I'll chime in on that. I'm very confident that this change is the correct direction Moya should take, but I think delaying this to Moya 13.0 is a healthy decision. I'm biased to my own design so if possible I would like other @Moya/core-team members input before making this change 😕

**Comment by ashfurrow** [comment]
I'll take a look next week and let you know what I think 👍 I trust that you'll all come to the right decision, though!

**Comment by freak4pc** [comment]
Is there a 11 -> 12 Major-Major CHANGELOG? Or a Migration Guide? Is one needed?

**Comment by SD10** [comment]
Yeah, there's not a lot of breaking changes but I believe we need a migration guide. I'm working on it now

**Comment by SD10** [comment]
Migration guide is in #1710

**Comment by ashfurrow** [comment]
@pedrovereza I took a look at the plan you laid out, and I think that makes sense 👍

**Comment by pedrovereza** [comment]
Thanks @SD10 ❤️ 

I feel this issue is solved then. We can revisit the discussion about #1126 / #1502 in a new issue for the first stable release of version 12.

Thanks again everyone for taking the time to discuss the release 🙏
